spring-cloud-kubernetes-integration-tests/run.sh

```diff
@@ -56,81 +65,23 @@ install_kind_release() {
     chmod +x "${KIND}"
 }
 
-setup_registry() {
-set -o errexit
-
-# create registry container unless it already exists
-reg_name='kind-registry'
-reg_port='5000'
-running="$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)"
-if [ "${running}" != 'true' ]; then
-  docker run \
-    -d --restart=always -p "${reg_port}:5000" --name "${reg_name}" \
-    registry:2
-fi
-
-# create a cluster with the local registry enabled in containerd
-cat <<EOF | "${KIND}" create cluster --config=-
-kind: Cluster
-apiVersion: kind.x-k8s.io/v1alpha4
-containerdConfigPatches:
-- |-
-  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:${reg_port}"]
-    endpoint = ["http://${reg_name}:${reg_port}"]
-EOF
-
-# connect the registry to the cluster network
-docker network connect "kind" "${reg_name}"
-
-# tell https://tilt.dev to use the registry
-# https://docs.tilt.dev/choosing_clusters.html#discovering-the-registry
-for node in $("${KIND}" get nodes); do
-  kubectl annotate node "${node}" "kind.x-k8s.io/registry=localhost:${reg_port}";
-done
-
-}
-
 main() {
     # get kind
     install_latest_kind
 
     # create a cluster
     cd $CURRENT_DIR
 
-    # create registry container unless it already exists
-	reg_name='kind-registry'
-	reg_port='5000'
-	running="$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)"
-	if [ "${running}" != 'true' ]; then
-		# If the container is not running but has exited remove it before trying to start it
-  		if [ "$(docker ps -aq -f status=exited -f name=${reg_name})" ]; then
-			# cleanup
-			docker rm "${reg_name}"
-  		fi
-
-  		docker run \
-    	-d --restart=always -p "${reg_port}:5000" --name "${reg_name}" \
-    	registry:2
-	fi
-
     #TODO what happens if cluster is already there????
-    "${KIND}" create cluster --config kind-config.yaml --loglevel=debug
-
-    # connect the registry to the cluster network
-	docker network connect "kind" "${reg_name}"
-
-	# tell https://tilt.dev to use the registry
-	# https://docs.tilt.dev/choosing_clusters.html#discovering-the-registry
-	for node in $("${KIND}" get nodes); do
-	  kubectl annotate node "${node}" "kind.x-k8s.io/registry=localhost:${reg_port}";
-	done
+    "${KIND}" create cluster --config=kind-config.yaml --loglevel=debug
 
     # set KUBECONFIG to point to the cluster
     kubectl cluster-info --context kind-kind
 
 	#setup nginx ingress
 #    kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/master/deploy/static/provider/kind/deploy.yaml
     kubectl apply -fhttps://raw.githubusercontent.com/kubernetes/ingress-nginx/12150e318b972a03fb49d827e6cabb8ef62247ef/deploy/static/provider/kind/deploy.yaml
+    sleep 5 # hold 5 sec so that the pods can be created
```

Comment:
This is what the `kubectl wait` command is for, I dont see why we need to sleep

Reply:
kubectl wait will exit immediately if there's no pod present in the cluster matching the label selector. this will be reproducible in a slower development environment. my old mac hardly bears a kind cluster sorry :(

Reply:
OK that makes sense thanks